02-feature-engineering.Rmd

```diff
@@ -1115,7 +1115,7 @@ recipe(rdg_spr ~ rdg_fall + ell, benchmarks) %>%
   bake(new_data = NULL)
 ```
 
-There are a few things to mention here. First, the data do look identical (minus the column names) to what we created by hand, so that's good news. However, we also used `starts_with` here, rather than just specifying something like `step_interact(~rdg_fall:ell)`. This is because after dummy coding, the column names change (as you can see in the above). By specifying `starts_with`, we are ensuring that we get all the interactions we need. If we forget this step, we end up with the wrong output, along with a warning.
+There are a few things to mention here. First, the data look identical (minus the column names) to what we created by hand, so that's good news. However, we also used `starts_with()` here, rather than just specifying something like `step_interact(~rdg_fall:ell)`. This is because after dummy coding, the column names change (as you can see in the above). By specifying `starts_with()`, we are ensuring that we get all the interactions we need. If we forget this step, we end up with the wrong output, along with a warning.
```

Comment:
I'm actually not seeing a warning message here. Might need to take a second look at this?

Reply:
Hmmm... weird. Okay, we'll have to look into that more.

Reply:
Just to be clear, you do **not** get a warning with the below?

```r
library(tidyverse)
library(tidymodels)

benchmarks <- read_csv(here::here("data", "benchmarks.csv"))

recipe(rdg_spr ~ rdg_fall + ell, benchmarks) %>% 
  step_dummy(all_nominal()) %>% 
  step_interact(~rdg_fall:ell) %>% 
  prep() %>% 
  bake(new_data = NULL)
```

I get

> Warning message:
Interaction specification failed for: ~rdg_fall:ell. No interactions will be created.

Reply:
Oh, I just realized the warning isn't showing up because of the setup chunk:

`knitr::opts_chunk$set(warning = FALSE)`

I'll set `warning = TRUE` for that chunk, that should solve it

Reply:
Actually, I don't think we should expect a warning there. What I was trying to say is that if we **don't** use `starts_with()` we get a warning. Maybe it just needs to be phrased differently?

Reply:
Oh you were right! Sorry, I didn't realize I actually had a chunk where the warning was supposed to be spit out.

Reply:
What you wrote originally makes sense. The code chunk where you use `step_interact(~rdg_fall:starts_with("ell"))` **does not** generate a warning and the chunk where you use `step_interact(~rdg_fall:ell)` **does** generate a warning. We just weren't seeing that warning before because all warnings were suppressed by default. The warning is showing up in the correct place now in the deploy preview
![Screen Shot 2020-11-10 at 10 53 01 AM](https://user-images.githubusercontent.com/32279312/98718401-40179380-2343-11eb-9ae0-30e6778f4ac5.png)

Reply:
oh sorry the comments aren't updating for me fast enough on Github! 😆  ok I think this should be ready to merge in!